agent.py

```diff
@@ -68,10 +84,19 @@ def learn(self, mem):
       # Calculate nth next state probabilities
       pns = self.online_net(next_states)  # Probabilities p(s_t+n, ·; θonline)
       dns = self.support.expand_as(pns) * pns  # Distribution d_t+n = (z, p(s_t+n, ·; θonline))
-      argmax_indices_ns = dns.sum(2).argmax(1)  # Perform argmax action selection using online network: argmax_a[(z, p(s_t+n, a; θonline))]
+
+      # Action Masking 
+      argmax_indices_masked = dns.sum(2)
```

Comment:
Can you just take the regular argmax here too without the mask? Then you might be able to get away whithout a second forward pass.